packages/expo-cli/src/commands/eas-build/utils/expoUpdates.ts

```diff
@@ -0,0 +1,514 @@
+  const username = await UserManager.getCurrentUsernameAsync();
+
+  const { exp } = getConfig(projectDir, { skipSDKVersionRequirement: true });
+
+  if (!exp.runtimeVersion && !exp.sdkVersion) {
+    throw new Error(
+      "Couldn't find either 'runtimeVersion' or 'sdkVersion' to configure 'expo-updates'. Please specify at least one of these properties under the 'expo' key in 'app.json'"
+    );
+  }
+
+  return { exp, username };
+}
+
+function isExpoUpdatesInstalled(projectDir: string) {
+  const packageJson = getPackageJson(projectDir);
+
+  return packageJson.dependencies && 'expo-updates' in packageJson.dependencies;
+}
+
+async function configureUpdatesIOSAsync(
+  projectDir: string,
+  exp: ExpoConfig,
+  username: string | null
+) {
+  const pbxprojPath = await getPbxprojPathAsync(projectDir);
+  const project = await getXcodeProjectAsync(pbxprojPath);
+  const bundleReactNative = await getBundleReactNativePhaseAsync(project);
+
+  if (!bundleReactNative.shellScript.includes(iOSBuildScript)) {
+    bundleReactNative.shellScript = `${bundleReactNative.shellScript.replace(
+      /"$/,
+      ''
+    )}${iOSBuildScript}\\n"`;
+  }
+
+  await fs.writeFile(pbxprojPath, project.writeSync());
+
+  await modifyExpoPlistAsync(projectDir, expoPlist => {
+    return IOSConfig.Updates.setUpdatesConfig(exp, expoPlist, username);
+  });
+}
+
+async function modifyExpoPlistAsync(projectDir: string, callback: (expoPlist: any) => any) {
+  const pbxprojPath = await getPbxprojPathAsync(projectDir);
+  const expoPlistPath = getExpoPlistPath(projectDir, pbxprojPath);
+
+  let expoPlist = {};
+
+  if (await fs.pathExists(expoPlistPath)) {
+    const expoPlistContent = await fs.readFile(expoPlistPath, 'utf8');
+    expoPlist = plist.parse(expoPlistContent);
+  }
+
+  const updatedExpoPlist = callback(expoPlist);
+
+  if (updatedExpoPlist === expoPlist) {
+    return;
+  }
+
+  const expoPlistContent = plist.build(updatedExpoPlist);
+
+  await fs.mkdirp(path.dirname(expoPlistPath));
+  await fs.writeFile(expoPlistPath, expoPlistContent);
+  await gitAddAsync(expoPlistPath, { intentToAdd: true });
+}
+
+export async function isUpdatesConfiguredIOSAsync(projectDir: string) {
+  if (!isExpoUpdatesInstalled(projectDir)) {
+    return true;
+  }
+
+  const { exp, username } = await getConfigurationOptionsAsync(projectDir);
+
+  const pbxprojPath = await getPbxprojPathAsync(projectDir);
+  const project = await getXcodeProjectAsync(pbxprojPath);
+  const bundleReactNative = await getBundleReactNativePhaseAsync(project);
+
+  if (!bundleReactNative.shellScript.includes(iOSBuildScript)) {
+    return false;
+  }
+
+  const expoPlistPath = getExpoPlistPath(projectDir, pbxprojPath);
+
+  if (!(await fs.pathExists(expoPlistPath))) {
+    return false;
+  }
+
+  const expoPlist = await fs.readFile(expoPlistPath, 'utf8');
+  const expoPlistData = plist.parse(expoPlist);
+
+  return isMetadataSetIOS(expoPlistData, exp, username);
+}
```

Comment:
Hmmm, maybe this should be moved to `@expo/config`? So far we've been using the `xcode` package only there. I think that we may want to leave it this way. 
@brentvatne any opinion on this?

Reply:
Generally, I think that it'd be better to move not only these functions but others (from this file) that interact with the native code (Android manifest, pbxproj, etc.)too. I might be wrong but this was the reason for creating `@expo/config`.